src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -37,6 +39,57 @@
  */
 public abstract class QueryBuilder<QB extends QueryBuilder> extends ToXContentToBytes implements Writeable<QB> {
 
+    public static enum Operator {
+        OR, AND;
+
+        public BooleanClause.Occur toMustOrShouldClause() {
+            switch (this) {
+                case OR:
+                    return BooleanClause.Occur.SHOULD;
+                case AND:
+                    return BooleanClause.Occur.MUST;
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
+
+        public static Operator readFrom(StreamInput in) throws IOException {
+            switch (in.readVInt()) {
+                case 0:
+                    return OR;
+                case 1:
+                    return AND;
+                default:
+                    throw new ElasticsearchException("Serialized operator needs to be either '1' or '0'");
+            }
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            switch (this) {
+                case OR:
+                    out.writeVInt(0);
+                case AND:
+                    out.writeVInt(1);
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
+
+        public static Operator fromString(String op) {
+            if ("or".equalsIgnoreCase(op)) {
```

Comment:
you could just loop over Operator.values here instead and compare operator.name() with `op`.